proto/services/tasks_svc/v1/task_template_svc.proto

```diff
@@ -0,0 +1,184 @@
+syntax = "proto3";
+
+package services.tasks_svc.v1;
+
+option go_package = "gen/proto/services/tasks-svc";
+
+///
+/// CreateTaskTemplate
+///
+
+message CreateTaskTemplateRequest {
+  string name = 1; // @gotags: validate:"required"
+  optional string description = 2;
+  optional string ward_id = 4;
+  repeated SubTask subtasks = 5; // When passed, every subtask is a seperate entity
+
+  message SubTask {
+    string name = 1;
+  }
+}
+
+message CreateTaskTemplateResponse {
+  string id = 1;
+}
+
+
+///
+/// GetAllTaskTemplates
+///
+
+message GetAllTaskTemplatesRequest {
+}
+
+message GetAllTaskTemplatesResponse {
+  repeated TaskTemplate templates = 1;
+
+  message TaskTemplate {
+    string id = 1;
+    string name = 2;
+    string description = 3;
+    bool is_public = 4;
+    string created_by = 6;
+    repeated SubTask subtasks = 7;
+
+    message SubTask {
+      string id = 1;
+      string task_template_id = 2;
+      string name = 3;
+    }
+  }
+}
+
+
+
+///
+/// DeleteTaskTemplate
+///
+
+message DeleteTaskTemplateRequest {
+  string id = 1; // @gotags: validate:"uuid4"
+}
+
+message DeleteTaskTemplateResponse {
+}
+
+
+///
+/// DeleteTaskTemplateSubTask
+///
+
+message DeleteTaskTemplateSubTaskRequest {
+  string id = 1; // @gotags: validate:"uuid4"
+}
+
+message DeleteTaskTemplateSubTaskResponse {
+}
+
+
+///
+/// UpdateTaskTemplate
+///
+
+message UpdateTaskTemplateRequest {
+  string id = 1; // @gotags: validate:"required"
+  optional string name = 2;
+  optional string description = 3;
+}
+
+message UpdateTaskTemplateResponse {
+}
+
+
+///
+/// UpdateTaskTemplateSubTask
+///
+
+message UpdateTaskTemplateSubTaskRequest {
+  string subtask_id = 1; // @gotags: validate:"required"
+  optional string name = 2;
+}
+
+message UpdateTaskTemplateSubTaskResponse {
+}
+
+
+///
+/// CreateTaskTemplateSubTask
+///
+
+message CreateTaskTemplateSubTaskRequest {
+  string task_template_id = 1; // @gotags: validate:"required"
+  string name = 2; // @gotags: validate:"required"
+}
+
+message CreateTaskTemplateSubTaskResponse {
+  string id = 1;
+}
+
+///
+/// GetAllTaskTemplatesByCreator
+///
+
+message GetAllTaskTemplatesByCreatorRequest {
+  string created_by = 1;
+  optional bool private_only = 2;
+}
+
+message GetAllTaskTemplatesByCreatorResponse {
+  repeated TaskTemplate templates = 1;
+
+  message TaskTemplate {
+    string id = 1;
+    string name = 2;
+    string description = 3;
+    bool is_public = 4;
+    repeated SubTask subtasks = 7;
+
+    message SubTask {
+      string id = 1;
+      string task_template_id = 2;
+      string name = 3;
+    }
+  }
+}
+
+
+///
+/// GetAllTaskTemplatesByWard
+///
+
+message GetAllTaskTemplatesByWardRequest {
```

Comment:
again, merge this with the ByCreator query into one with optionals

Reply:
will be noted

Reply:
#807